src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -720,6 +720,110 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     </EmbeddedResource>
   </ItemDefinitionGroup>
 
+  <!--
+    ========================
+    DeterministicSourcePaths
+    ========================
+    
+    Unless specified otherwise enable deterministic source root when building deterministically on CI server, but not for local builds.
+    In order for the debugger to find source files when debugging a locally built binary the PDB must contain original, unmapped local paths.
+  -->
+  <PropertyGroup>
+    <DeterministicSourcePaths Condition="'$(DeterministicSourcePaths)' == '' and '$(Deterministic)' == 'true' and '$(ContinuousIntegrationBuild)' != ''">true</DeterministicSourcePaths>
+  </PropertyGroup>
+
+  <!--
+    ==========
+    SourceRoot
+    ==========
+    
+    All source files of the project are expected to be located under one of the directories specified by SourceRoot item group.
+    This targets collects all SourceRoots from various sources (InitializeSourceRootDependsOn targets).
+    
+    If DeterministicSourcePaths is true it calculates deterministic mapped paths that can be used by CoreCompile targets
+    to initialize deterministic path mapping specific to the compiler.
+    
+    Include the final backslash in the path of each SourceRoot item.
+    
+    Extensibility: Add target to InitializeSourceRootDependsOn that populates SourceRoot item group.
+  -->
+
+  <Target Name="InitializeSourceRoot"
+          DependsOnTargets="$(InitializeSourceRootDependsOn)">
+    <ItemGroup>
+      <_InvalidSourceRoot Include="@(SourceRoot)" Condition="!HasTrailingSlash('%(SourceRoot.Identity)')" />
+    </ItemGroup>
+
+    <Error Text="SourceRoot paths are required to end with a slash or backslash: @(_InvalidSourceRoot, ', ')"
```

Comment:
Localization?

Reply:
How do I localize an error message in .targets file?

Reply:
Given that there's other error tasks in the file and I searched the visual studio drops for all occurrences of Microsoft.Common.CurrentVersion.targets but did not find anything, it's probably safe to say that, because it's always been not localized, it can remain not localized.